sky/clouds/service_catalog/data_fetchers/fetch_vast.py

```diff
@@ -0,0 +1,67 @@
+"""A script that generates the Vast Cloud catalog. """
+
+# pylint: disable=assignment-from-no-return
+import csv
+import json
+import re
+import sys
+
+from vastai_sdk import VastAI
+
+
+def create_instance_type(obj):
+    stubify = lambda x: re.sub(r'\s', '_', x)
+    return '{}x-{}-{}'.format(obj['num_gpus'], stubify(obj['gpu_name']),
+                              obj['cpu_cores'])
+
+
+def dot_get(d, key):
+    for k in key.split('.'):
+        d = d[k]
+    return d
+
+
+# InstanceType and gpuInfo are basically just stubs
+# so that the dictwriter is happy without weird
+# code.
+mapped_keys = (('gpu_name', 'InstanceType'), ('gpu_name', 'AcceleratorName'),
+               ('num_gpus', 'AcceleratorCount'), ('cpu_cores', 'vCPUs'),
+               ('gpu_total_ram', 'MemoryGiB'), ('search.totalHour', 'Price'),
+               ('geolocation', 'Region'), ('gpu_name', 'GpuInfo'),
+               ('search.totalHour', 'SpotPrice'))
```

Comment:
our "spot" price (interruptible) is a live bidding market with all the volatility that comes with that. Contracts can be bought, outbid, and resold similar to futures markets. 

The multi-hour caching time makes this very stale and so we are communicating the higher price here, which is still often the lowest in the industry.

This was a compromise made because skypilot doesn't support the pricing system.

Reply:
I see. cc @Michaelvll for a look here